ledger/internal/eval_test.go

```diff
@@ -266,13 +281,116 @@ func TestCowStateProof(t *testing.T) {
 	blockErr[13] = noBlockErr
 	err = apply.StateProof(stateProofTx, atRound, c0, validate)
 	require.Contains(t, err.Error(), "no block")
+	delete(blockErr, 13)
+
+	// check the happy flow - we should fail only on crypto
+	atRound = 800
+	spHdr = bookkeeping.BlockHeader{}
+	spHdr.CurrentProtocol = version
+	blocks[basics.Round(2*config.Consensus[version].StateProofInterval)] = spHdr
+
+	votersHdr := bookkeeping.BlockHeader{}
+	votersHdr.CurrentProtocol = version
+	stateproofTracking := bookkeeping.StateProofTrackingData{
+		StateProofVotersCommitment:  []byte{0x1}[:],
+		StateProofOnlineTotalWeight: basics.MicroAlgos{Raw: 5},
+	}
+	votersHdr.StateProofTracking = make(map[protocol.StateProofType]bookkeeping.StateProofTrackingData)
+	votersHdr.StateProofTracking[protocol.StateProofBasic] = stateproofTracking
+
+	blocks[basics.Round(config.Consensus[version].StateProofInterval)] = votersHdr
+	atRoundBlock = bookkeeping.BlockHeader{}
+	atRoundBlock.CurrentProtocol = version
+	blocks[atRound] = atRoundBlock
+
+	stateProofTx.Message.LastAttestedRound = 2 * config.Consensus[version].StateProofInterval
+	stateProofTx.StateProof.SignedWeight = 100
+	c0.SetStateProofNextRound(basics.Round(2 * config.Consensus[version].StateProofInterval))
 
-	// fall through to no err
-	validate = false
 	err = apply.StateProof(stateProofTx, atRound, c0, validate)
-	require.NoError(t, err)
+	require.Contains(t, err.Error(), "crypto error")
+}
+
+func TestCowStateProof(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	var spType protocol.StateProofType
+	var stateProof stateproof.StateProof
+	atRound := basics.Round(600)
+	var validate bool
+	msg := stateproofmsg.Message{}
+
+	accts0 := ledgertesting.RandomAccounts(20, true)
+	blocks := make(map[basics.Round]bookkeeping.BlockHeader)
+	blockErr := make(map[basics.Round]error)
+	ml := mockLedger{balanceMap: accts0, blocks: blocks, blockErr: blockErr}
+	c0 := makeRoundCowState(
+		&ml, bookkeeping.BlockHeader{}, config.Consensus[protocol.ConsensusFuture],
+		0, ledgercore.AccountTotals{}, 0)
+	ml.stateProofVerification = make(map[basics.Round]*ledgercore.StateProofVerificationData)
+
+	spType = protocol.StateProofType(1234) // bad stateproof type
+	stateProofTx := transactions.StateProofTxnFields{
+		StateProofType: spType,
+		StateProof:     stateProof,
+		Message:        msg,
+	}
+	err := apply.StateProof(stateProofTx, atRound, c0, validate)
+	require.ErrorIs(t, err, apply.ErrStateProofTypeNotSupported)
 
-	// 100% coverage
```

Comment:
no more 100% coverage?

Reply:
just thought that the comments doesn't help much :)
If you think there is an edge case that we didn't cover, I will happily test it